src/runtime/internal/dom.ts

```diff
@@ -5,7 +5,9 @@ export function append(target: Node, node: Node) {
 }
 
 export function insert(target: Node, node: Node, anchor?: Node) {
-	target.insertBefore(node, anchor || null);
+	if (target != document) {
```

Comment:
silently doing nothing seems like the worst of all possible behaviors to me. I'd either append the new element of just leave things as is. this doesn't seem like a very big problem, so I'm not sure it's worth the extra code and complication. if we did want this then you should probably do `!==` and write a test, but I'm rather hesitant about this addition

Reply:
Hi @benmccann, thanks for looking at this. I wasn't sure about the right way to approach it, but I'd be happy to revise. My limited understanding is the insert function creates the instructions to mount the elements into the target (in this case the document). The top level document can only have 1 element/child (e.g <html>), so it throws this error. Assuming we use `!==` it does silently prevent this, but would there ever be a situation where we would want to insert on the document? Maybe there's a better approach though, can you help me understand what I'd need to do to "append the new element?" Thank you!